src/main/kotlin/org/rust/ide/intentions/MergeIfsIntention.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsBinaryExpr
+import org.rust.lang.core.psi.RsExpr
+import org.rust.lang.core.psi.RsIfExpr
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.LogicOp
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.operatorType
+
+class MergeIfsIntention : RsElementBaseIntentionAction<MergeIfsIntention.Context>() {
+
+    override fun getText(): String = "Merge 'if's"
```

Comment:
The first thing I thought was that this intention merges all nested expressions

```rust
// before
if a {
    if b {
        if c {
            return;
        }
    }
}
    
// after
if a && b && c {
    return;
}
```

I think something like "Merge with the nested `if` expression" is more clear.